app/src/main/java/com/kodeco/android/countryinfo/ui/components/CountryInfoScreen.kt

```diff
@@ -0,0 +1,83 @@
+package com.kodeco.android.countryinfo.ui.components
+
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.tooling.preview.Preview
+import com.kodeco.android.countryinfo.api.CountryAPIService
+import com.kodeco.android.countryinfo.models.Country
+import com.kodeco.android.countryinfo.models.CountryFlags
+import com.kodeco.android.countryinfo.models.CountryName
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import retrofit2.Response
+
+sealed class CountryInfoState {
+    object Loading : CountryInfoState()
+    data class Success(val countries: List<Country>) : CountryInfoState()
+    data class Error(val error: Throwable) : CountryInfoState()
+}
+
+@Composable
+fun CountryInfoScreen(service: CountryAPIService) {
+
+    var infoState: CountryInfoState by remember { mutableStateOf(CountryInfoState.Loading) }
+    Surface {
+        when (val currentState = infoState) {
+            is CountryInfoState.Loading -> Loading()
+            is CountryInfoState.Success -> CountryInfoList(currentState.countries)
+            is CountryInfoState.Error -> CountryErrorScreen(currentState.error, onTryAgain = {
+                infoState = CountryInfoState.Loading
+            })
+        }
+    }
+
+    if (infoState == CountryInfoState.Loading) {
+        LaunchedEffect(Unit) {
+            launch {
+                delay(1000)
+                infoState = try {
+                    val countriesResponse = service.getAllCountries()
+
+                    if (countriesResponse.isSuccessful) {
+                        CountryInfoState.Success(countriesResponse.body()!!)
+                    } else {
+                        CountryInfoState.Error(Throwable("Request Failed - ${countriesResponse.message()}"))
+                    }
+                } catch (exception: Exception) {
+                    CountryInfoState.Error(exception)
+                }
```

Comment:
If you remove the `Response` wrapper as I mentioned elsewhere you could simplify this code here too:

```suggestion
                infoState = try {
                    CountryInfoState.Success(service.getAllCountries()) {
                } catch (exception: Exception) {
                    CountryInfoState.Error(exception)
                }
```

Reply:
Nice solution. Implemented and pushed. Thank you.